### PR TITLE
Add example for plotting a bihistogram

### PR DESCRIPTION
**Pull request by i-jey**
## PR summary
Adds an example to the documentation for plotting a bihistogram (see the closed PR #27180 for prior discussion)

## PR checklist

- [NA] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [X] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [X] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [NA] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [X] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Comment by scottshambaugh** [comment]
image

I think this example is well done and useful! Only suggestion is that I think it would make more sense visually when comparing datasets like this to have a constant bin width rather than constant number of bins, such as 
```
bin_width=0.25
bins=np.arange(min(dataset1), max(dataset1) + bin_width, bin_width)
ax.hist(dataset1, bins=bins, label="Dataset 1")
```
image

**Comment by i-jey** [comment]
That's a good catch, totally agree. I've added that in, thanks for taking a look!

File: galleries/examples/statistics/histogram_bihistogram.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+"""
+==========
```
Comment (scottshambaugh) [inline galleries/examples/statistics/histogram_bihistogram.py:2]: Title and section delimiter lines need to be the same length as their text or the docs build system will throw a fit. Lines 2, 4, and 17.

**Comment by scottshambaugh** [comment]
The built docs look good to me, thanks!

**Comment by scottshambaugh** [comment]
@i-jey thanks and congratulations on your first contribution to Matplotlib! 🎉

We hope to see you again.

**Pull request merged**